channel.go

```diff
@@ -70,6 +70,7 @@ func (m *ChannelMember) UnmarshalJSON(data []byte) error {
 	}
 
 	removeFromMap(m.ExtraData, *m)
+	flattenExtraData(m.ExtraData)
```

Comment:
I am not sure how json parsing works in go but shouldn't we flip the order here?

```suggestion
	flattenExtraData(m.ExtraData)
	removeFromMap(m.ExtraData, *m)
```